elm-src/NixJSON.elm

```diff
@@ -0,0 +1,150 @@
+module NixJSON exposing (Gson, gnull, nixFromGson, gsonDecoder)
+
+import Dict exposing (Dict, empty, foldr, filter)
+import Json.Decode as Decode
+
+type Gson
+    = GString String
+    | GBool Bool
+    | GInt Int
+    | GFloat Float
+    | GList (List Gson)
+    | GDict (Dict String Gson)
+    | GNull
+
+gnull : Gson
+gnull = GNull
+
+nixFromGson : Gson -> String
+nixFromGson gson =
+    case gson of
+        GNull ->
+            "null"
+
+        GString x ->
+            nix_string x
+
+        GBool True ->
+            "true"
+
+        GBool False ->
+            "false"
+
+        GInt x ->
+            toString x
+
+        GFloat x ->
+            toString x
+
+        GList list ->
+            nix_list list
+
+        GDict dict ->
+            if ((Dict.member "_type" dict) && ((Dict.get "_type" dict) == Just (GString "literalExample")) && (Dict.member "text" dict)) then
+                case Dict.get "text" dict of
+                    Just (GString txt) ->
+                        txt
+
+                    otherwise ->
+                        "Literal example failed to produce text"
+            else
+                let
+                    entries =
+                        Dict.toList dict
+                in
+                    if List.length entries == 0 then
+                        "{ }"
+                    else
+                        String.concat
+                            (List.concat
+                                [ [ "{\n" ]
+                                , indent
+                                    (List.concat
+                                        (List.map nix_enc_dict entries)
+                                    )
+                                , [ "}" ]
+                                ]
+                            )
+
+
+nix_string : String -> String
```

Comment:
It took me a while to realize this, but what's with the `snake_case` functions? This feels very weird. Some sort of disambiguation between exposed and non-exposed functions? I'd say that doesn't really make sense - whether a function is exposed or not doesn't make a difference to _how_ you'd use it within the defining module.

I feel like I'm missing something?

Reply:
I think it wasn't clear to me how I should  name my functions, and then effectively picked at random 😳 I'll clean these up :) thank you!